src/core_plugins/kibana/server/lib/manage_uuid.js

```diff
@@ -0,0 +1,73 @@
+import uuid from 'node-uuid';
+import Promise from 'bluebird';
+import { hostname } from 'os';
+import { join as pathJoin } from 'path';
+import { readFile as readFileCallback, writeFile as writeFileCallback } from 'fs';
+
+const serverHostname = hostname();
+const FILE_ENCODING = 'utf8';
+
+export default async function manageUuid(server) {
+  const config = server.config();
+  const serverPort = server.info.port;
+  const fileName = `${serverHostname}:${serverPort}`;
+  const uuidFile = pathJoin(config.get('path.data'), fileName);
```

Comment:
Actually, I do run multiple Kibanae out of the same run directory for testing. I just specify the config file, port and server name on the command line:
```
./bin/kibana --config config/kibana.dev.yml --server.port 5611 --server.name kbn32
```

Good point about the port change. I think the solution to this is allow the user to set the UUID themselves in config if they need a port change. This could also be a solution to keep the metrics contained together across an upgrade.